### PR TITLE
drivers/counter/counter_cmos.c: implement counter with PC AT "CMOS" RTC

### PR DESCRIPTION
**Pull request by ghost**
Enable use of the PC/AT "CMOS" RTC as a simple 1Hz counter.

Signed-off-by: Charles E. Youse <charles.youse@intel.com>

### CODE REVIEW

**Review by tyrantlizard77 — APPROVED** [review]
Not intimately familiar with this hardware, but the code looks sound.

**Comment by rakons** [comment]
I am really not convinced for the name. As the CMOS sounds like the technology used for manufacturing ( complementary metal–oxide–semiconductor) and for the software we should stay away from the implementation details (what if the same RTC would be implemented in TTL technology?).

I would rather vote for ```counter_pcat``` or even ```counter_pcrtc``` that sounds more descriptive.

**Comment by nashif** [comment]
hmm, that license issue is curious, something is wrong. the license of file is apache.

**Comment by gnuless** [comment]
> I am really not convinced for the name. As the CMOS sounds like the technology used for manufacturing 

You're right that it's a misnomer, but "the CMOS" is what this component has been called since the PC/AT arrived on the scene, 30+ years ago.

**Comment by gnuless** [comment]
> hmm, that license issue is curious, something is wrong. the license of file is apache.

It's probably picking up on my mention of Hinnant's algorithm as "public domain".. I'll reword.

**Comment by gnuless** [comment]
Added a test case with the last push.

File: drivers/counter/counter_cmos.c [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+{
+	ARG_UNUSED(dev);
+
+	return 0;
+}
+
+static const struct counter_driver_api api = {
+	.read = read
+};
+
```
Comment (nordic-krch) [inline drivers/counter/counter_cmos.c]: would be good to add `counter_config_info` to the device then at least frequency and max_top_value and direction can be determined through counter API.

Comment (gnuless) [reply drivers/counter/counter_cmos.c]: @nordic-krch Good idea. It's a great project for a summer intern... so I've assigned one. :)

**Pull request merged**